Remove misc. references to undefined Kconfig symbols

ulfalizer opened the pull request:
I'm about to add [a CI check](https://github.com/zephyrproject-rtos/ci-tools/pull/21) for references to undefined Kconfig symbols. This gets rid of some actual issues from the symbol whitelist.

ulfalizer committed:
doc: tests: Remove ref. to undefined sym. CONFIG_ADC_QMSI_SAMPLE_WIDTH

CONFIG_ADC_QMSI_SAMPLE_WIDTH was removed in commit acc40d990a9 ("adc:
remove adc_qmsi driver"). It's only used as an example in the
documentation, but might as well use an existing symbol instead.

Signed-off-by: Ulf Magnusson <Ulf.Magnusson@nordicsemi.no>

ulfalizer committed:
arch: arc: Remove ref. to undefined sym. CONFIG_FIRQ_NO_LPCC

This Kconfig symbol has never been defined in the Git history.

Signed-off-by: Ulf Magnusson <Ulf.Magnusson@nordicsemi.no>

ulfalizer committed:
ext: hal: nordic: Remove ref. to undefined CONFIG_IEEE802154_NRF5_RAW

Might have been overlooked in commit b52c0f24a6 ("net/ieee802154: Make
RAW mode generic") and commit f4664f1d9a ("drivers/ieee802154: Cleaning
up CMakeLists.txt").

Signed-off-by: Ulf Magnusson <Ulf.Magnusson@nordicsemi.no>

ulfalizer committed:
toolchain: gcc: Remove ref. to undefined sym. CONFIG_ISA_ARM

This Kconfig symbol was removed in commit cb04620fbd ("remove make/
directory").

Signed-off-by: Ulf Magnusson <Ulf.Magnusson@nordicsemi.no>

ulfalizer committed:
toolchain: Remove ref. to undefined sym. CONFIG_ISA_THUMB

This symbol has never been defined in the Git history.

Signed-off-by: Ulf Magnusson <Ulf.Magnusson@nordicsemi.no>

ulfalizer committed:
arduino_101_mcuboot: Remove CONFIG_UART_QMSI_0_BAUDRATE - moved to DTS

UART_QMSI_0_BAUDRATE was removed by commit 17c6456678 ("drivers/uart:
Use dts to set uart priorities for QMSI driver").

There's a reference to CONFIG_UART_QMSI_1_BAUDRATE as well, but getting
rid of it requires documentation updates.

Signed-off-by: Ulf Magnusson <Ulf.Magnusson@nordicsemi.no>

dbkinder reviewed (approved):
+1 for docs

nashif commented on doc/guides/test/sanitycheck.rst:
```diff
@@ -341,7 +341,7 @@ extra_configs: <list of extra configurations>
           test_resolution_6:
             extra_configs:
-              - CONFIG_ADC_QMSI_SAMPLE_WIDTH=6
+              - CONFIG_LOG_DEFAULT_LEVEL=3
```
how did an ADC related Kconfig turn into a LOG_LEVEL?

ulfalizer replied:
I just picked that random symbol that exists. What's a better example I could use?

ulfalizer replied:
@nashif *ping*

Note that this is in documentation.

ulfalizer replied:
@nashif 

Changed it to this. Is that fine?

```
common:
  tags: drivers adc
tests:
  test:
    depends_on: adc
  test_async:
    extra_configs:
      - CONFIG_ADC_ASYNC=y
```

ulfalizer committed:
doc: dts: Clarify CONFIG_SRAM documentation

There is no symbol called CONFIG_SRAM. Mention CONFIG_SRAM_SIZE and
CONFIG_SRAM_BASE_ADDRESS instead, and also mention that the DT_* values
they're generated from.

This gets rid of a false-positive undefined Kconfig symbol reference
too.

Signed-off-by: Ulf Magnusson <Ulf.Magnusson@nordicsemi.no>
